### PR TITLE
Do not record underlying calls in provenance

### LINKED ISSUE #740: Disable low-level provenance recording when using CLI
We currently record provenance from API calls which aren't informative from the CLI. 

Just add ``record_provenance=False`` to API calls from CLI module when #558 is merged.

### PR DESCRIPTION
**Pull request by hyanwong**
Fixes #740  - also adds provenance recording to generate-ancestors CLI call

### CODE REVIEW

File: tsinfer/cli.py [hunk 1]
```diff
@@ -177,13 +178,19 @@ def run_generate_ancestors(args):
...
-    tsinfer.generate_ancestors(
+    ancestors = tsinfer.generate_ancestors(
         sample_data,
         progress_monitor=args.progress,
-        path=ancestors_path,
         num_flush_threads=args.num_flush_threads,
         num_threads=args.num_threads,
+        record_provenance=False,
     )
+    new_ancestors = ancestors.copy(path=ancestors_path)
```
Comment (jeromekelleher) [inline tsinfer/cli.py:188]: This is a big cost just for provenance, and will lead to a huge pause after the progress bar is done. I don't think it's worth it.

Maybe just skip this one and open an issue to track? There's no useful options in generate ancestors to record anywy

**Review by jeromekelleher — COMMENTED** [review]
Looks good, but the match-ancestors stuff is a big performance and usabilty regression.

File: tsinfer/cli.py (hunk 1 above)
Comment (hyanwong) [reply tsinfer/cli.py:188]: Good point: I hadn't thought about the perf issues of making a copy. I assume there's no way to add provenance to the existing ancestors file without copying?

As you say, not worth spending time over, if we are re-doing the pipeline anyway, so we can just open an issue to track.

**Review by jeromekelleher — APPROVED** [review]
LGTM

**Pull request merged**